src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -332,6 +335,7 @@ public bool IsSet(AvaloniaProperty property)
                 }
 
                 subscription = source
+                    .Do(_ => VerifyAccess())
```

Comment:
This will never throw unless someone switches which scheduler is being used (i.e. gets rid of the `AvaloniaScheduler`). Binding updates are already forced onto the UI thread by the `AvaloniaScheduler` on line 314 in this file.